src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java

```diff
@@ -224,9 +288,65 @@ void tesGenerateAcspApplicationFilingWithBlankAddresses() throws Exception {
+    }
+
+
+    @Test
+    void tesGenerateAcspApplicationFilingWithNoNamesAndId() throws Exception {
+        //initTransactionPaymentLinkMocks();
+        //initGetPaymentMocks();
+
+        setACSPDataDtoWithoutNamesandId();
+        acspDataDto.setCorrespondenceAddresses(buildNullCorrespondenceAddress());
+        acspDataDto.setBusinessAddress(buildNullBusinessAddress());
+        when(acspService.getAcsp(any(), any())).thenReturn(Optional.of(acspDataDto));
+        when(transactionService.getTransaction(PASS_THROUGH_HEADER, TRANSACTION_ID)).thenReturn(transaction);
 
+        var response = filingsService.generateAcspApplicationFiling(ACSP_ID, TRANSACTION_ID, PASS_THROUGH_HEADER);
+        //Assertions.assertEquals("100", response.getCost());
+        //Assertions.assertEquals(PAYMENT_REFERENCE.toUpperCase(), response.getData().get("payment_reference"));
+        //Assertions.assertEquals(PAYMENT_METHOD.toUpperCase(), response.getData().get("payment_method"));
+        Assertions.assertNotNull(response.getData().get("item"));
+        Assertions.assertNotNull(response.getData().get("presenter"));
+        Assertions.assertNull(((Presenter)response.getData().get("presenter")).getFirstName());
+        Assertions.assertNull(((Presenter)response.getData().get("presenter")).getLastName());
+        Assertions.assertNull(((Presenter)response.getData().get("presenter")).getUserId());
+        Assertions.assertNull(response.getDescriptionIdentifier());
+        Assertions.assertNull(response.getDescription());
     }
 
+    @Test
+    void tesGenerateAcspApplicationFilingWithNoEmail() throws Exception {
+        //initTransactionPaymentLinkMocks();
+        //initGetPaymentMocks();
+
+        setACSPDataDtoWithoutemailaddress();
+        when(acspService.getAcsp(any(), any())).thenReturn(Optional.of(acspDataDto));
+        when(transactionService.getTransaction(PASS_THROUGH_HEADER, TRANSACTION_ID)).thenReturn(transaction);
+
+        var response = filingsService.generateAcspApplicationFiling(ACSP_ID, TRANSACTION_ID, PASS_THROUGH_HEADER);
+        Assertions.assertNull(((ACSP) response.getData().get("data")).getEmail());
+    }
+
+
+    @Test
+    void tesGenerateAcspApplicationFilingWithCompanyDetailsWithoutCompanyNameAndNumber() throws Exception {
+        //initTransactionPaymentLinkMocks();
+        //initGetPaymentMocks();
+
+        setACSPDataDtoWithCompanyDetailsButNoCompanyNumberAndName();
+        when(acspService.getAcsp(any(), any())).thenReturn(Optional.of(acspDataDto));
+        when(transactionService.getTransaction(PASS_THROUGH_HEADER, TRANSACTION_ID)).thenReturn(transaction);
+
+        var response = filingsService.generateAcspApplicationFiling(ACSP_ID, TRANSACTION_ID, PASS_THROUGH_HEADER);
+        Assertions.assertNull(((ACSP) response.getData().get("data")).getCompanyName());
+        Assertions.assertNull(((ACSP) response.getData().get("data")).getCompanyNumber());
+    }
+
+
```

Comment:
Can you remove commented lines from the tests.

Reply:
I wil have to add these back when I bring the payment bits back